File: LeetCode/1980/golang/lc1980_test.go
package lc1980

import "testing"

func isValidAnswer(candidate string, nums []string) bool {
	if len(candidate) != len(nums[0]) {
		return false
	}
	for _, num := range nums {
		if num == candidate {
			return false
		}
	}
	return true
}

func Test_findDifferentBinaryString(t *testing.T) {
	testcases := []struct {
		nums []string
	}{
		{[]string{"01", "10"}},
		{[]string{"00", "01"}},
		{[]string{"111", "011", "001"}},
	}

	for i, tc := range testcases {
		actual := findDifferentBinaryString(tc.nums)
		if !isValidAnswer(actual, tc.nums) {
			t.Errorf("Testcase %02d failed: got %s which does not match %v", i+1, actual, tc.nums)
		}
	}
}

File: LeetCode/1980/python/lc1980_test.py
import unittest

from lc1980 import Solution

class Test_findDifferentBinaryString(unittest.TestCase):
    def test(self):
        s = Solution()
        self.assertIn(s.findDifferentBinaryString(["01", "10"]), ["00", "11"])
        self.assertIn(s.findDifferentBinaryString(["00", "01"]), ["10", "11"])
        self.assertIn(s.findDifferentBinaryString(["111", "011", "001"]), ["000", "010", "100", "101", "110"])

if __name__ == "__main__":
    unittest.main()
